internal/controller/dnsrecord_controller.go

```diff
@@ -178,6 +179,7 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		dnsRecord.SetStatusZoneDomainName("")
 		dnsRecord.SetStatusZoneID("")
 		dnsRecord.SetStatusDomainOwners(nil)
+		controllerutil.RemoveFinalizer(dnsRecord.GetDNSRecord(), DNSRecordFinalizer)
```

Comment:
This doesn't seem right. With this change, we will not wait for the ` dnsRecord.GetStatus().ProviderEndpointsRemoved()` to be true

Reply:
yes, you're right. I've already undone this locally but not pushed it up yet.